ksnp4/4.0/Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM ubuntu:jammy as app
+ 
+# metadata
+LABEL base.image="ubuntu:jammy"
+LABEL dockerfile.version="1"
+LABEL software="kSNP4"
+LABEL software.version="4.0"
+LABEL description="K-mer-based phylogenetic inference"
+LABEL website="https://sourceforge.net/projects/ksnp/"
+LABEL license="https://github.com/kissake/kSNP4/blob/%40/THE%20BSD%20OPENSOURCE%20LICENSE.pdf"
+LABEL maintainer="Jeremiah Nisbet"
+LABEL maintainer.email="ksnp-dev@kissake.net"
+
+
+RUN apt-get update && apt-get install -y python2 \
+  wget tcsh unzip make gcc build-essential && \
+  cd /opt &&\
+  wget https://github.com/stamatak/standard-RAxML/archive/v8.2.12.tar.gz && \
+  tar xvfz v8.2.12.tar.gz && \
+  cd /opt/standard-RAxML-8.2.12 && \
+  make -f Makefile.SSE3.PTHREADS.gcc && \
+  rm *.o && \
+  ln -s $(pwd)/raxmlHPC-PTHREADS-SSE3 /usr/bin/raxml && \
+  chmod a+x ./usefulScripts/*.* && \
+  sed -i 's/^\$raxmlExecutable =*.*/\$raxmlExecutable = "raxml";/' ./usefulScripts/ProteinModelSelection.pl && \
+  cd /opt/ && \
+  wget "https://sourceforge.net/projects/ksnp/files/kSNP4 Linux package.zip" && \
+  unzip "kSNP4 Linux package.zip" && \
+  mv "kSNP4 Linux package/kSNP4pkg/" /usr/local/ && \
+  rm -r "kSNP4 Linux package.zip" && \
+  mkdir data
```

Comment:
These lines are the "installation lines". The lines that you have are basically the same ones from version 3.1, which was 2 years ago. We've learned a lot since then.

Most of the dockerfiles in this repo split installations into different 'RUN' layers. I've tried to split it out, but there may in a typo in here somewhere.
```
RUN apt-get update && apt-get -y --no-install-recommends install \
  python2 \
  wget \
  tcsh \
  unzip \
  make \
  gcc \
  build-essential && \
  apt-get autoclean && rm -rf /var/lib/apt/lists/*

# install RAxML
RUN  wget https://github.com/stamatak/standard-RAxML/archive/v${RAXMLVER}.tar.gz && \
  tar xvfz v${RAXMLVER}.tar.gz && \
  cd /standard-RAxML-${RAXMLVER} && \
  make -f Makefile.SSE3.PTHREADS.gcc && \
  rm *.o && \
  ln -s $(pwd)/raxmlHPC-PTHREADS-SSE3 /usr/bin/raxml && \
  chmod a+x ./usefulScripts/*.* && \
  sed -i 's/^\$raxmlExecutable =*.*/\$raxmlExecutable = "raxml";/' ./usefulScripts/ProteinModelSelection.pl

# install kSNP4
RUN wget "https://sourceforge.net/projects/ksnp/files/kSNP4 Linux package.zip" && \
  unzip "kSNP4 Linux package.zip" && \
  mv "kSNP4 Linux package/kSNP4pkg/" /usr/local/ && \
  rm -r "kSNP4 Linux package.zip" && \
  mkdir /data
 ```

This helps readability.

Reply:
Also, does the raxml in apt-get work for kSNP4? (It might make things easier)

```
RUN apt-get update && apt-get -y --no-install-recommends install \
  python2 \
  wget \
  tcsh \
  unzip \
  make \
  gcc \
  raxml \
  build-essential && \
  apt-get autoclean && rm -rf /var/lib/apt/lists/*
  ```